frontend/app/assets/javascripts/spree/frontend/checkout.js.coffee

```diff
@@ -10,3 +10,8 @@ Spree.disableSaveOnClick = ->
 
 Spree.ready ($) ->
   Spree.Checkout = {}
+
+  terms_checkbox = ($ 'form.edit_order #accept_term_and_condition')
+  terms_checkbox.change( () ->
+    ($ 'form.edit_order :submit').prop('disabled', !this.checked)
+  )
```

Comment:
Usually using jQuery's `closest does a good job at finding the nearest form, which would save you from specifying it with an ephemeral class.